test/test_binary_ufuncs.py

```diff
@@ -1183,6 +1183,34 @@ def test_binary_ops_with_scalars(self, device):
                         self.assertEqual(expected, python_op(first, second))
                         self.assertEqual(expected, torch_op(first, second))
 
+    # Tests that the function and its (array-accepting) reference produce the same
+    #   values on a range of tensors, including empty tensors, scalar tensors,
+    #   1D tensors and a large 2D tensor
+    @onlyOnCPUAndCUDA
+    @ops(binary_ufuncs)
+    def test_reference_numerics(self, device, dtype, op):
```

Comment:
I would call this "test_reference_numerics" because those tests in test_unary_ufuncs.py generate their own massive inputs; this is more like "test_reference_samples"?

What do you think is the future for a test like this? Is it worth comparing both with samples and large random tensors? If so, should we also compare the unary pwise ops vs their references using samples? If so, should we put this in test_ops.py and filter it to the binary and unary ufuncs?

Reply:
I have no answer to this yet. @mattip thoughts?